geode-core/src/main/java/org/apache/geode/internal/cache/wan/AbstractGatewaySender.java

```diff
@@ -1298,6 +1301,26 @@ public boolean removeFromTempQueueEvents(Object tailKey) {
     }
   }
 
+  public boolean markAsDuplicateInTempQueueEvents(Object tailKey) {
+    synchronized (queuedEventsSync) {
+      Iterator<TmpQueueEvent> itr = tmpQueuedEvents.iterator();
+      while (itr.hasNext()) {
+        TmpQueueEvent event = itr.next();
+        if (tailKey.equals(event.getEvent().getTailKey())) {
+          if (logger.isDebugEnabled()) {
```

Comment:
Check for if debug enabled once outside this loop and use a final variable here to gate the logging.

Reply:
Updated.